e2e/e2e.sh

```diff
@@ -18,6 +18,8 @@ set -o errexit
 set -o pipefail
 set -o nounset
 
+export KOPS_STATE_STORE=${KOPS_STATE_STORE:-"state-${KOPS_CLUSTER_NAME}"}
```

Comment:
If we want this, then we should check if `KOPS_CLUSTER_NAME` is set and error out if it doesn't. Otherwise, the export is going to fail if `KOPS_CLUSTER_NAME` isn't set due to `nounset`.

I'm wondering though if a default `KOPS_STATE_STORE` makes sense here -- after all, the user needs to make sure it exists, so my thinking is that we could just as well expect them to specify the store name (as we do right now).

Maybe I missed something about your intentions, so please let me know.

Reply:
I don't think the user has to make sure this exists on their own (I've never created it myself), but it is checked as required and was erroring previously to this default being set here:
```
    e2e_test.go:57: missing required environment variable(s): [KOPS_STATE_STORE]
```
An alternative is that we set it to a derived default like this in the Go code closer to where it's actually needed.

Reply:
You're right, the variable shouldn't need to be set for the end-to-end tests because [we're making sure it is set](https://github.com/digitalocean/digitalocean-cloud-controller-manager/blob/47731f5a0c87b53ebb06171e6851b8233e679eca/e2e/cluster_test.go#L78) prior to calling `setup_cluster.sh`. It is only needed when someone invokes that script directly (which I happen to do every now and then).

I think it's okay to just exclude `KOPS_STATE_STORE` from [the set of variables checked for existence](https://github.com/digitalocean/digitalocean-cloud-controller-manager/blob/47731f5a0c87b53ebb06171e6851b8233e679eca/e2e/e2e_test.go#L51) then in the Go test. The [check we have in `setup_cluster.sh`](https://github.com/digitalocean/digitalocean-cloud-controller-manager/blob/47731f5a0c87b53ebb06171e6851b8233e679eca/e2e/scripts/setup_cluster.sh#L26) suffices, and so we can drop this extra export. WDYT?

Reply:
That makes sense to me :+1: